model/vsstree.py

```diff
@@ -0,0 +1,141 @@
+
+
+class VSSNode(Node):
+    description = None
+    uuid = None
+    type: VSSType
+    data_type = VSSDataType.NONE
+
+    unit = None
+    min = None
+    max = None
+    enum = None
+    deprecation = None
+
+    aggregate = False
+    child_type = None
+    child_properties = 0
+    prop_name = None
+    prop_type = None
+    prop_format = None
+    prop_unit = None
+    prop_value = None
+
+    default_value = None
+    value = None
+
+    instances = None
+
+    def __init__(self, name, vss_type: VSSType, vss_uuid, vss_description, parent=None, children=None):
+        super().__init__(name, parent, children)
+        self.description = vss_description
+        self.type = vss_type
+        self.uuid = vss_uuid
+
+    def is_private(self):
+        node = self
+        while node:
+            if node.name == "Private":
+                return True
+            node = node.parent
+        return False
+
+    @staticmethod
+    def case_conversion(source, style: StringStyle):
+        if style == StringStyle.ALPHANUM_CASE:
+            return stringcase.alphanumcase(source)
+        elif style == StringStyle.CAMEL_CASE:
+            return camel_case(source)
+        elif style == StringStyle.CAMEL_BACK:
+            return camel_back(source)
+        elif style == StringStyle.CAPITAL_CASE:
+            return stringcase.capitalcase(source)
+        elif style == StringStyle.CONST_CASE:
+            return stringcase.constcase(source)
+        elif style == StringStyle.LOWER_CASE:
+            return stringcase.lowercase(source)
+        elif style == StringStyle.PASCAL_CASE:
+            return stringcase.pascalcase(source)
+        elif style == StringStyle.SENTENCE_CASE:
+            return stringcase.sentencecase(source)
+        elif style == StringStyle.SNAKE_CASE:
+            return stringcase.snakecase(source)
+        elif style == StringStyle.SPINAL_CASE:
+            return stringcase.spinalcase(source)
+        elif style == StringStyle.TITLE_CASE:
+            return stringcase.titlecase(source)
+        elif style == StringStyle.TRIM_CASE:
+            return stringcase.trimcase(source)
+        elif style == StringStyle.UPPER_CASE:
+            return stringcase.uppercase(source)
+        else:
+            return source
+
+    def qualified_name(self, separator="/", style=StringStyle.NONE):
+        name = VSSNode.case_conversion(self.name, style)
+
+        path = self
+        while not path.is_root:
+            path = path.parent
+            node_name = path.name
+            node_name = VSSNode.case_conversion(node_name, style)
+
+            name = "%s%s%s" % (node_name, separator, name)
+        return name
+
+    def is_orphan(self):
+        if self.type == VSSType.BRANCH or self.type == VSSType.RBRANCH:
+            return self.is_leaf
+        return False
+
+    def is_deprecated(self):
+        return self.deprecation is not None
+
+    def supports_unit(self):
```

Comment:
A lot of these functions I see are not used in the current source code.  I think you have some idea for future usage?  Let's discuss those details, please.

Reply:
Not all VSS leafs have a unit defined. This might make a difference for generating code or other sorts of definitions from the model. Thus, I added little helper functions like this to delegate unit support to the VSSNode instead of handling it in the generators.